cmd/containerd-shim-runhcs-v1/exec_hcs.go

```diff
@@ -612,29 +585,26 @@ func (he *hcsExec) waitForExit() {
 		const processCopyTimeout = time.Second * 1
 
 		time.Sleep(processCopyTimeout)
-		he.sl.Lock()
-		defer he.sl.Unlock()
-		if he.stdout != nil || he.stderr != nil {
+		// Close the process to cancel any reads to stdout or stderr.
+		he.pCloseOnce.Do(func() {
 			logrus.WithFields(logrus.Fields{
 				"tid": he.tid,
 				"eid": he.id,
 			}).Warn("hcsExec::waitForExit - timed out waiting for ioRelay to complete")
 
-			if he.stdout != nil {
-				he.stdout.Close()
-				he.stdout = nil
-			}
-			if he.stderr != nil {
-				he.stderr.Close()
-				he.stderr = nil
-			}
-		}
+			he.p.Close()
```

Comment:
This is safe to call multiple times. Was this just to only write this log message if the process wasn't already closed?

Reply:
Yes.